tests/common/auth_test.go

```diff
@@ -736,6 +736,121 @@ func TestAuthRoleList(t *testing.T) {
 	})
 }
 
+func TestAuthJWTExpire(t *testing.T) {
+	testRunner.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1, AuthToken: defaultAuthToken}))
+	defer clus.Close()
+	cc := testutils.MustClient(clus.Client())
+	testutils.ExecuteUntil(ctx, t, func() {
+		require.NoErrorf(t, setupAuth(cc, []authRole{testRole}, []authUser{rootUser, testUser}), "failed to enable auth")
+		testUserAuthClient := testutils.MustClient(clus.Client(WithAuth(testUserName, testPassword)))
+
+		require.NoError(t, testUserAuthClient.Put(ctx, "foo", "bar", config.PutOptions{}))
+		// wait an expiration of my JWT token
+		<-time.After(3 * time.Second)
+
+		// e2e test will generate a new token while
+		// integration test that re-uses the same etcd client will refresh the token on server failure.
```

Comment:
Could we validate that we refreshed token?

Reply:
Yeah, it is feasible if 

1. etcd server log level is debug level
2. capture the etcd server log output

https://github.com/etcd-io/etcd/blob/4da39e4b1e4bc6028f69b29171e82149a72e5b37/server/auth/store.go#L339-L343